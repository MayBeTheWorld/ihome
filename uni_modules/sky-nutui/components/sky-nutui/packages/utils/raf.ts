const _window = window as any;
function requestAniFrame() {
  if (typeof _window !== 'undefined') {
    return (
      _window.requestAnimationFrame ||
      _window.webkitRequestAnimationFrame ||
      function (callback: Function) {
        _window.setTimeout(callback, 1000 / 60);
      }
    );
  } else {
    return function (callback: Function) {
      setTimeout(callback, 1000 / 60);
    };
  }
}

export default requestAniFrame();
