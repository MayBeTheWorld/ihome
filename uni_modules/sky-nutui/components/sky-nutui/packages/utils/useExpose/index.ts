import { getCurrentInstance } from 'vue';

export function useExpose(apis: Record<string, any>) {
  const instance = getCurrentInstance();
  if (instance) {
    Object.assign(instance.proxy, apis);
  }
}
