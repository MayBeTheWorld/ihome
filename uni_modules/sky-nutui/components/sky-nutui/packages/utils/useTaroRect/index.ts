/**
  获取元素的大小及其相对于视口的位置，等价于 Element.getBoundingClientRect。
  width 宽度	number
  height 高度	number
  top	顶部与视图窗口左上角的距离	number
  left	左侧与视图窗口左上角的距离	number
  right	右侧与视图窗口左上角的距离	number
  bottom	底部与视图窗口左上角的距离	number
 */

import { Ref, unref } from 'vue';
function isWindow(val: unknown): val is Window {
  return val === window;
}
export const useTaroRect = (elementRef: (Element | Window | any) | Ref<Element | Window | any>, self: any): any => {
  
    return new Promise((resolve) => {
        const query = uni.createSelectorQuery().in(self);
        query
          .select(elementRef)
          .boundingClientRect((rect: any) => {
              resolve(rect)
          })
          .exec();
    })

};
