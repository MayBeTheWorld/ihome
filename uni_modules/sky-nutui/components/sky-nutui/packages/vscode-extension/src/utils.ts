export const kebabCase = (str: string): string => {
  str = str.replace(str.charAt(0), str.charAt(0).toLocaleLowerCase());
  return str.replace(/([a-z])([A-Z])/g, (_, p1, p2) => p1 + '-' + p2.toLowerCase());
};

export const camelize = (str: string): string => {
  return str.replace(/-(\w)/g, (_: any, p: string) => p.toUpperCase());
};

export const bigCamelize = (str: string): string => {
  return camelize(str).replace(str.charAt(0), str.charAt(0).toUpperCase());
};
