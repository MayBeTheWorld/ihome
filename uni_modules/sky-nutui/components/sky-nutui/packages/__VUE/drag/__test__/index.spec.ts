import { mount } from '@vue/test-utils';
import Drag from '../index.vue';
import { triggerDrag, sleep } from '../../../utils/unit';

test('should render default slot correctly', () => {
  const wrapper = mount(Drag, {
    props: {
      attract: true,
      direction: 'y',
      boundary: { top: 10, left: 12, right: 20, bottom: 40 }
    },
    slots: {
      default: () => 'Custom Message'
    }
  });
  expect(wrapper.find('.nut-drag').html()).toMatchSnapshot();
});

test('touch move', async () => {
  const wrapper: any = mount(Drag, {
    slots: {
      default: () => 'Custom Message'
    }
  });
  triggerDrag(wrapper, 20, 100);
  await sleep(1000);
  expect(wrapper.element.style.left).toBe('20px');
});
test('should render attract when touch', async () => {
  const wrapper: any = mount(Drag, {
    props: {
      attract: true,
      direction: 'x'
    },
    slots: {
      default: () => `custom`
    }
  });
  triggerDrag(wrapper, 200, 0);
  await sleep(1000);
  expect(wrapper.element.style.left).toBe('375px');
  triggerDrag(wrapper, 100, 0);
  await sleep(1000);
  expect(wrapper.element.style.left).toBe('0px');
});
test('boundary touchMove', async () => {
  const wrapper: any = mount(Drag, {
    props: {
      boundary: { top: 360, left: 10, right: 100, bottom: 300 }
    },
    slots: {
      default: () => `custom`
    }
  });
  triggerDrag(wrapper, 200, 200);
  await sleep(1000);
  expect(wrapper.element.style.left).toBe('200px');
  expect(wrapper.element.style.top).toBe('360px');
});
