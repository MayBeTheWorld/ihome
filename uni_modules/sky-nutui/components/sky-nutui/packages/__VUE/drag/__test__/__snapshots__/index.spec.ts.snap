// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default slot correctly 1`] = `"<view class=\\"nut-drag\\">Custom Message</view>"`;
